src/alg_traits.jl

```diff
@@ -295,3 +295,34 @@ initialization.
 Defaults to false.
 """
 allows_late_binding_tstops(alg::AbstractODEAlgorithm) = false
+
+"""
+    $(TYPEDSIGNATURES)
+
+Trait declaration for whether an algorithm creates a cache on calling `init` which stores the tolerances.
+
+Defaults to false.
+
+If this returns true, `cache_get_abstol` and `cache_get_reltol` must be implemented for the appropriate cache type.
+"""
+cache_stores_tolerances(alg::AbstractSciMLAlgorithm) = false
+
+"""
+    $(TYPEDSIGNATURES)
+
+Get the `abstol` for the solve stored in the cache. Required for caches of algorithms for
+which `cache_stores_tolerances(alg) == true`.
+"""
+function cache_get_abstol(cache)
+    error("Cache does not store the `abstol`.")
+end
+
+"""
+    $(TYPEDSIGNATURES)
+
+Get the `reltol` for the solve stored in the cache. Required for caches of algorithms for
+which `cache_stores_tolerances(alg) == true`.
+"""
+function cache_get_reltol(cache)
+    error("Cache does not store the `reltol`.")
```

Comment:
I thought we talked about passing it?

Reply:
Didn't you also mention having a trait and throwing an error message? Right now it takes a keyword argument and falls back to this if not provided.

Reply:
That said I'm more than happy to remove this trait since solvers have to individually opt-in to doing initialization anyway.

Reply:
Let's simplify and just make it required.